Google storage provider with existing bucket

**Is this a request for help?**:
YES

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

possibly a BUG

**Version of Helm and Kubernetes**:
1.11.7

**Which chart**:
0.4.4 artifactory-ha

**What happened**:

We are trying to activate google storage binary provider. As part of the setup we provision the bucket via terraform and then deploy artifactory ha helm chart.  I have set the <bucketExists> field to true in the artifactory-binarystore.yaml as suggested [here](https://www.jfrog.com/confluence/display/RTF/Configuring+the+Filestore#ConfiguringtheFilestore-GoogleStorageBinaryProvider). However when I deploy the pod in the primary statefulset does not stabilise. The following messages are in the logs

[art-init] [INFO ] (o.a.a.f.t.j.g.GoogleJets3tBinaryProvider:94) - Bucket exists = false
[ERROR] (o.a.a.f.t.j.g.GoogleJets3tBinaryProvider:274) - Failed to create bucket bucket-name on GCS --> BucketNameUnavailable:The requested bucket name is not available. The bucket namespace is shared by all users of the system. Please select a different name and try again.

**What you expected to happen**:

Should respect the bucketExists flag and not try to recreate it

**How to reproduce it** (as minimally and precisely as possible):

Enable google-storage provider and pass it existing bucket name and interoperability API credentials and try to deploy it

--- a/stable/artifactory-ha/CHANGELOG.md
+++ b/stable/artifactory-ha/CHANGELOG.md
@@ -1,6 +1,9 @@
 # JFrog Artifactory-ha Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [0.12.14] - Aprl 12, 2019
+* Added support for `bucketExists` flag for googleStorage
+
 ## [0.12.13] - Aprl 07, 2019
 * Add support for providing the Artifactory license as a parameter
 

--- a/stable/artifactory-ha/Chart.yaml
+++ b/stable/artifactory-ha/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: artifactory-ha
 home: https://www.jfrog.com/artifactory/
-version: 0.12.13
+version: 0.12.14
 appVersion: 6.9.1
 description: Universal Repository Manager supporting all major packaging formats,
   build tools and CI servers.

--- a/stable/artifactory-ha/README.md
+++ b/stable/artifactory-ha/README.md
@@ -613,6 +613,7 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.persistence.googleStorage.identity`    | Google Storage service account id   |                              |
 | `artifactory.persistence.googleStorage.credential`  | Google Storage service account key  |                              |
 | `artifactory.persistence.googleStorage.path`        | Google Storage path in bucket       | `artifactory-ha/filestore`   |
+| `artifactory.persistence.googleStorage.bucketExists`| Google Storage bucket exists therefore does not need to be created.| `false`             |
 | `artifactory.persistence.awsS3.bucketName`          | AWS S3 bucket name                     | `artifactory-ha`             |
 | `artifactory.persistence.awsS3.endpoint`            | AWS S3 bucket endpoint                 | See https://docs.aws.amazon.com/general/latest/gr/rande.html |
 | `artifactory.persistence.awsS3.region`              | AWS S3 bucket region                   |                              |

--- a/stable/artifactory-ha/templates/artifactory-binarystore.yaml
+++ b/stable/artifactory-ha/templates/artifactory-binarystore.yaml
@@ -120,6 +120,7 @@ data:
             <identity>{{ .Values.artifactory.persistence.googleStorage.identity }}</identity>
             <credential>{{ .Values.artifactory.persistence.googleStorage.credential }}</credential>
             <path>{{ .Values.artifactory.persistence.googleStorage.path }}</path>
+            <bucketExists>{{ .Values.artifactory.persistence.googleStorage.bucketExists }}</bucketExists>
         </provider>
     </config>
 {{- end }}

--- a/stable/artifactory-ha/templates/nginx-deployment.yaml
+++ b/stable/artifactory-ha/templates/nginx-deployment.yaml
@@ -66,9 +66,6 @@ spec:
               - '/bin/sh'
               - '-c'
               - >
-                {{- if .Values.nginx.customConfigMap }}
-                cp -Lrf /tmp/nginx.conf /etc/nginx/nginx.conf;
-                {{- end }}
                 {{- if .Values.nginx.customArtifactoryConfigMap }}
                 cp -Lrf /tmp/conf.d/artifactory*.conf /etc/nginx/conf.d/artifactory.conf;
                 {{- else }}
@@ -123,7 +120,8 @@ spec:
         {{- end }}
         {{- if .Values.nginx.customConfigMap }}
         - name: nginx-config
-          mountPath: "/tmp/"
+          mountPath: "/etc/nginx/nginx.conf"
+          subPath: "nginx.conf"
         {{- end }}
         {{- if .Values.nginx.customArtifactoryConfigMap }}
         - name: artifactory-nginx-config

--- a/stable/artifactory-ha/values.yaml
+++ b/stable/artifactory-ha/values.yaml
@@ -375,6 +375,7 @@ artifactory:
       identity:
       credential:
       path: "artifactory-ha/filestore"
+      bucketExists: "false"
     ## For artifactory.persistence.type aws-s3
     ## IMPORTANT: Make sure S3 `endpoint` and `region` match! See https://docs.aws.amazon.com/general/latest/gr/rande.html
     awsS3:

--- a/stable/artifactory-ha/values.yaml
+++ b/stable/artifactory-ha/values.yaml
@@ -375,7 +375,7 @@ artifactory:
       identity:
       credential:
       path: "artifactory-ha/filestore"
-      bucketExists: "false"
+      bucketExists: false
     ## For artifactory.persistence.type aws-s3
     ## IMPORTANT: Make sure S3 `endpoint` and `region` match! See https://docs.aws.amazon.com/general/latest/gr/rande.html
     awsS3:

--- a/stable/artifactory/CHANGELOG.md
+++ b/stable/artifactory/CHANGELOG.md
@@ -1,6 +1,9 @@
 # JFrog Artifactory Chart Changelog
 All changes to this chart will be documented in this file.
 
+## [7.13.12] - Aprl 12, 2019
+* Added support for `bucketExists` flag for googleStorage
+
 ## [7.13.11] - Aprl 07, 2019
 * Add support for providing the Artifactory license as a parameter
 

--- a/stable/artifactory/Chart.yaml
+++ b/stable/artifactory/Chart.yaml
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: artifactory
 home: https://www.jfrog.com/artifactory/
-version: 7.13.11
+version: 7.13.12
 appVersion: 6.9.1
 description: Universal Repository Manager supporting all major packaging formats,
   build tools and CI servers.

--- a/stable/artifactory/README.md
+++ b/stable/artifactory/README.md
@@ -507,6 +507,7 @@ The following table lists the configurable parameters of the artifactory chart a
 | `artifactory.persistence.googleStorage.identity`    | Google Storage service account id   |                              |
 | `artifactory.persistence.googleStorage.credential`  | Google Storage service account key  |                              |
 | `artifactory.persistence.googleStorage.path`        | Google Storage path in bucket       | `artifactory/filestore`   |
+| `artifactory.persistence.googleStorage.bucketExists`| Google Storage bucket exists therefore does not need to be created.| `false`             |
 | `artifactory.persistence.awsS3.bucketName`          | AWS S3 bucket name                     | `artifactory`             |
 | `artifactory.persistence.awsS3.endpoint`            | AWS S3 bucket endpoint                 | See https://docs.aws.amazon.com/general/latest/gr/rande.html |
 | `artifactory.persistence.awsS3.region`              | AWS S3 bucket region                   |                              |

--- a/stable/artifactory/templates/artifactory-binarystore.yaml
+++ b/stable/artifactory/templates/artifactory-binarystore.yaml
@@ -46,6 +46,7 @@ data:
             <identity>{{ .Values.artifactory.persistence.googleStorage.identity }}</identity>
             <credential>{{ .Values.artifactory.persistence.googleStorage.credential }}</credential>
             <path>{{ .Values.artifactory.persistence.googleStorage.path }}</path>
+            <bucketExists>{{ .Values.artifactory.persistence.googleStorage.bucketExists }}</bucketExists>
         </provider>
     </config>
 {{- end }}

--- a/stable/artifactory/templates/nginx-deployment.yaml
+++ b/stable/artifactory/templates/nginx-deployment.yaml
@@ -66,9 +66,6 @@ spec:
               - '/bin/sh'
               - '-c'
               - >
-                {{- if .Values.nginx.customConfigMap }}
-                cp -Lrf /tmp/nginx.conf /etc/nginx/nginx.conf;
-                {{- end }}
                 {{- if .Values.nginx.customArtifactoryConfigMap }}
                 cp -Lrf /tmp/conf.d/artifactory*.conf /etc/nginx/conf.d/artifactory.conf;
                 {{- else }}
@@ -123,7 +120,8 @@ spec:
         {{- end }}
         {{- if .Values.nginx.customConfigMap }}
         - name: nginx-config
-          mountPath: "/tmp/"
+          mountPath: "/etc/nginx/nginx.conf"
+          subPath: "nginx.conf"
         {{- end }}
         {{- if .Values.nginx.customArtifactoryConfigMap }}
         - name: artifactory-nginx-config

--- a/stable/artifactory/values.yaml
+++ b/stable/artifactory/values.yaml
@@ -300,6 +300,7 @@ artifactory:
       identity:
       credential:
       path: "artifactory/filestore"
+      bucketExists: "false"
     ## For artifactory.persistence.type aws-s3
     ## IMPORTANT: Make sure S3 `endpoint` and `region` match! See https://docs.aws.amazon.com/general/latest/gr/rande.html
     awsS3:

--- a/stable/artifactory/values.yaml
+++ b/stable/artifactory/values.yaml
@@ -300,7 +300,7 @@ artifactory:
       identity:
       credential:
       path: "artifactory/filestore"
-      bucketExists: "false"
+      bucketExists: false
     ## For artifactory.persistence.type aws-s3
     ## IMPORTANT: Make sure S3 `endpoint` and `region` match! See https://docs.aws.amazon.com/general/latest/gr/rande.html
     awsS3:

